source/parser.hera

```diff
@@ -4069,7 +4072,18 @@ ImplicitExportSpecifier
 Declaration
   HoistableDeclaration
   ClassDeclaration
-  LexicalDeclaration
+  LexicalDeclaration:d ->
+    if (d.thisAssignments?.length)
+      return {
+        ...d,
+        children: [...d.children, ...d.splices, ";", ...d.thisAssignments]
```

Comment:
Will this throw if splices is undefined? Maybe worth building up an array and protecting splices and thisAssignments pushing with ifs.